src/main/java/com/thealgorithms/misc/FourSumProblem.java

```diff
@@ -0,0 +1,59 @@
+package com.thealgorithms.misc;
+
+import java.util.*;
+
+/**
+ * Solution for The Four Sum Problem.
+ *
+ * @author mouss3abbb
+ */
+public final class FourSumProblem {
```

Comment:
It does not make any sense to have objects of this class (because it has only `static` methods).

```suggestion
public final class FourSumProblem {
    private FourSumProblem() {
    }
```